frontend/src/pages/NewRun.tsx

```diff
@@ -136,53 +157,95 @@ class NewRun extends Page<{}, NewRunState> {
+                columns={this.pipelineSelectorColumns}
+                emptyMessage='No pipelines found. Upload a pipeline and then try again.'
+                initialSortColumn={PipelineSortKeys.CREATED_AT}
+                resourceToRow={this._resourceToRow}
+                selectionChanged={this._pipelineSelectionChanged.bind(this)} />
             </DialogContent>
             <DialogActions>
               <Button id='cancelPipelineSelectionBtn' onClick={() => this._pipelineSelectorClosed(false)} color='secondary'>
                 Cancel
               </Button>
               <Button id='usePipelineBtn' onClick={() => this._pipelineSelectorClosed(true)}
-                color='secondary' disabled={!unconfirmedDialogPipelineId}>
+                color='secondary' disabled={!unconfirmedSelectedPipeline}>
                 Use this pipeline
               </Button>
             </DialogActions>
           </Dialog>
 
+          <Dialog open={experimentSelectorOpen}
+            classes={{ paper: css.selectorDialog }}
+            onClose={() => this._experimentSelectorClosed(false)}
+            PaperProps={{ id: 'experimentSelectorDialog' }}>
+            <DialogContent>
+            <SelectorList {...this.props}
+                title='Choose an experiment'
+                listApi={Apis.experimentServiceApi.listExperiment.bind(Apis)}
+                columns={this.experimentSelectorColumns}
+                emptyMessage='No experiments found. Create an experiment and then try again.'
+                initialSortColumn={ExperimentSortKeys.CREATED_AT}
+                resourceToRow={this._resourceToRow}
+                selectionChanged={this._experimentSelectionChanged.bind(this)} />
+            </DialogContent>
+            <DialogActions>
+              <Button id='cancelExperimentSelectionBtn' onClick={() => this._experimentSelectorClosed(false)} color='secondary'>
+                Cancel
+              </Button>
+              <Button id='useExperimentBtn' onClick={() => this._experimentSelectorClosed(true)}
+                color='secondary' disabled={!unconfirmedSelectedExperiment}>
+                Use this experiment
+              </Button>
+            </DialogActions>
+          </Dialog>
+
           <Input label='Run name' required={true} onChange={this.handleChange('runName')}
             autoFocus={true} value={runName} />
           <Input label='Description (optional)' multiline={true}
             onChange={this.handleChange('description')} value={description} />
 
-          {experimentName && (
-            <div>
-              <div>This run will be associated with the following experiment</div>
-              <Input label='Experiment' onChange={this.handleChange('experimentName')}
-                disabled={true} value={experimentName} />
-            </div>
-          )}
+          <div>
+            <div>This run will be associated with the following experiment</div>
```

Comment:
Should we change this message? Maybe "Select this run's experiment"?

Reply:
Maybe the description should be removed entirely? We don't say anything specific for the other fields

Reply:
I'd vote to keep it, since the UI still doesn't explain what experiments are anywhere, it's likely the least understood concept on that form.

Reply:
If that's the case, then I think we should keep the original wording as it's more explanatory

Reply:
SGTM.